corehq/apps/users/bulkupload.py

```diff
@@ -36,17 +36,30 @@ class UserUploadError(Exception):
 required_headers = set(['username'])
 allowed_headers = set([
     'data', 'email', 'group', 'language', 'name', 'password', 'phone-number',
-    'uncategorized_data', 'user_id', 'is_active', 'location-sms-code',
+    'uncategorized_data', 'user_id', 'is_active', 'location_code',
 ]) | required_headers
+old_headers = {
+    # 'old_header_name': 'new_header_name'
+    'location-sms-code': 'location_code'
+}
 
 
 def check_headers(user_specs):
+    messages = []
     headers = set(user_specs.fieldnames)
 
+    # Backwards warnings
+    for (old_name, new_name) in old_headers.iteritems():
+        if old_name in headers:
+            messages.append(
+                _("'The column header '{old_name}' is illegel, please use '{new_name}' instead.").format(
```

Comment:
This seems a little harsh. Maybe use the word "deprecated" or "no longer in use" instead? Either way, it's spelt 'illegal'.